### ISSUE TITLE
Add constraints to GA or other exploration method

### ISSUE THREAD

**Issue by yululeah** [gama-platform/gama#3072]
Hi, I am trying to run the genetic algorithm in GAMA, and I wonder how can I add some constraints in the constructor or experiment?
Like, I want the sum of my parameter 1 and 2 is less than 10, or something else.

Thanks!

**Comment by yululeah** [comment]
Or can I call GAMA from python, R, etc. to run GA with constraints?

**Comment by benoitgaudou** [comment]
Dear,
As far as I know, there is no way to introduce constraints among parameters.
Possible solutions could be to use the OpenMole software to design
experiment plan.
There could be a solution using R (the call of GAMA from R had worked and
is being repaired by @Arthur BRUGIERE ***@***.***> ).
Cheers

Benoit

**Comment by RoiArthurB** [comment]
Hi,

TL;DR: No official solution does exist yet for your problem, but we're working on it!

Here's the documentation of the genetic algorithm in gama's batch mode : https://gama-platform.github.io/wiki/Statements#genetic

Currently, the model exploration rely on external tools (like OpenMole https://openmole.org/GAMA.html ) and very early wrapper (like gamar : https://r-and-gama.github.io/gamar/ for R, the first stable version should be released in some weeks). These solutions are developed in parallel of the improvement of the headless part of GAMA.

**Comment by yululeah** [comment]
Thanks a lot!
